### ISSUE TITLE
[Bug][a11y]Focus on last selected tab in tab tray with a11y turned on

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#16424]
## Prerequisites
Talkback should be enabled

## Steps to reproduce
1. Open Google from the Top Sites section
2. Tap the Tab counter (The Tabs Tray opens)
3. Tap the "New tab" button
4. Navigate to Twitter.com
5. Tap the Tab counter (The Tabs Tray opens)

### Expected behavior
Step 2: 
In List view: The google.com tab should be selected after opening the tabs tray 
In grid view: The google.com tab should be selected after opening the tabs tray

Step 5:
In List view: Twitter.com tab should be selected after opening the tabs tray 
In grid view: Twitter.com tab should be selected after opening the tabs tray 

### Actual behavior
Step 2: 
In List view: The "Save to collection" button is focused
In grid view: The Tab counter icon is selected

Step 5:
In List view: Google.com is selected (previously opened website)
In grid view: The Tab counter icon is selected

### Device information

* Android device: 
• Google Pixel 3a (Android 11)
• Huawei Mate 20 Lite (Android 10)
• OnePlus A3 (Android 6.0.1)

* Fenix version: Nightly Build 201106

### Notes
❗ Previously done in #9640

**Comment by Mugurell** [comment]
I think this is an older regression from #10925 which reversed the order of items in the layout but [here](https://github.com/mozilla-mobile/fenix/blob/172977f6987066357144992fdcedc86d67c7de87/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L161) we still search for the selected tab counting from the beginning.

**Comment by Mugurell** [comment]
My solves the issue for the incorrect tab being focused and read by Talkback
while using list view
For grid view there is a separate issue - https://github.com/mozilla-mobile/fenix/issues/16417 that shows no browser tab being selected when opening tabs tray.

**Comment by ebalazs-sv** [comment]
Verified as fixed on Nightly 201111 05:01 (Build #2015774955) GV 84 from 11/11 with Pixel 2 (Android 9).
The focus is on the last selected tab in tab tray with a11y turned on in list view mode.
For grid view mode, there is a separate issue: https://github.com/mozilla-mobile/fenix/issues/16417.

I will close this issue and remove the qa:needed label.
